components/applications-service/pkg/storage/postgres/services.go

```diff
@@ -229,6 +235,77 @@ func (db *Postgres) GetDisconnectedServices(thresholdMinutes int32) ([]*storage.
 	return convertComposedServicesToStorage(services), err
 }
 
+// DeleteDisconnectedServices deletes any service records where the time
+// elapsed since the last event is greater than the specified thresholdMinutes.
+// After deleting the services, it also removes all supervisor, service_group,
+// and deployment records that are not referenced by any service record because
+// the application treats these things as emergent properties of groups of
+// services so they shouldn't exist if there are no associated services. This
+// also prevents the need for a second cleanup operation to delete these
+// things.
+func (db *Postgres) DeleteDisconnectedServices(thresholdMinutes int32) ([]*storage.Service, error) {
+
+	tx, err := db.DbMap.Begin()
```

Comment:
Also why changing the way we were doing transactions here? Did you find a limitation or just preference?

Reply:
I wanted to stick to what gorp exposes so you only need the docs for one library to understand what's going on.